### PR TITLE
Add a command to fill the username and password + TOTP.

### PR DESCRIPTION
**Pull request by m4mnux**
In some 2FA environments the password field must be filled
not only with the password but with the concatenation
of the passwword and the TOTP.

This patch allows to set up a key combination to do
do the job.

Signed-off-by: Miguel Martín <m4mnux@gmail.com>

### CODE REVIEW

File: keepassxc-browser/content/keepassxc-browser.js [hunk 1]
```diff
@@ -1529,7 +1550,24 @@ kpxc.fillIn = function(combination, onlyPassword, suppressWarnings) {
         }
 
         if (pField) {
-            kpxc.setValueWithChange(pField, kpxc.credentials[combination.loginId].password);
+            let password = kpxc.credentials[combination.loginId].password;
+            if (appendTOTP) {
+                if (kpxc.credentials[commbination.loginId].stringFields && kpxc.credentials[combination.loginId].stringFields.length > 0) {
+                    const stringFields = kpxc.credentials[combination.loginId].stringFields;
```
Comment (varjolintu) [inline keepassxc-browser/content/keepassxc-browser.js:1556]: Haven't tried your PR yet, but try to get rid of this duplicate code. It's almost identical to the stuff above. Maybe a separate function (inside the function?) for it?

Comment (m4mnux) [reply keepassxc-browser/content/keepassxc-browser.js]: I refactored the code to use a new _'kpxc.getTOTP'_ function. I hope this looks good to you

File: keepassxc-browser/content/keepassxc-browser.js [hunk 2]
```diff
@@ -1491,7 +1504,17 @@ kpxc.fillIn = function(combination, onlyPassword, suppressWarnings) {
         }
         if (pField) {
             pField.setAttribute('type', 'password');
-            kpxc.setValueWithChange(pField, kpxc.credentials[0].password);
+            let password = kpxc.credentials[0].password;
+            let totp = '';
```
Comment (varjolintu) [inline keepassxc-browser/content/keepassxc-browser.js:1508]: No need to define the variable here. You can just use `const totp = kpxc.getTOTP(kpxc.credentials[0]);` inside the if sentence.

File: keepassxc-browser/content/keepassxc-browser.js [hunk 3]
```diff
@@ -1529,7 +1552,17 @@ kpxc.fillIn = function(combination, onlyPassword, suppressWarnings) {
         }
 
         if (pField) {
-            kpxc.setValueWithChange(pField, kpxc.credentials[combination.loginId].password);
+            let password = kpxc.credentials[combination.loginId].password;
+            let totp = '';
```
Comment (varjolintu) [inline keepassxc-browser/content/keepassxc-browser.js:1556]: Same here.

**Comment by varjolintu** [comment]
I'm not sure if this functionality is something we even need to the extension. This is the first time I've heard some site even use that kind of login mechanism. Making a totally new feature for just one or two sites.. I'm not sure if it's worth it.

**Comment by m4mnux** [comment]
How do I confirm the selection? there's no button and 'enter' key does nothing

**Comment by varjolintu** [comment]
There's a Confirm button in the DOM dialog. You can also press <kbd>c</kbd> as the help text says.

**Comment by m4mnux** [comment]
Pressing `c` works for me but I am not able to see any help text or any green Confirm button. :(
Not sure if the problem is the page itself or the browser version (Tried with Firefox 75.0 and Chromium 81.0.4044.92)

**Comment by varjolintu** [comment]
This is how it looks for me before confirmation:
Screen Shot 2020-05-05 at 12 05 48

**Pull request closed without merging**